steam-client/src/main/java/com/avenga/steamclient/steam/client/SteamClient.java

```diff
@@ -212,16 +227,26 @@ private GCMessage getGCPacketMessage(PacketMessage packetMessage) {
         return new GCMessage(gcClientMessage.getBody());
     }
 
+    private void handleServiceMethodBody(PacketMessage packetMessage, ClientMessageProtobuf body) {
+        if (body.getBody() instanceof SteammessagesPlayerSteamclient.CPlayer_GetLastPlayedTimes_Response.Builder) {
+            var playedTimesResponse = (SteammessagesPlayerSteamclient.CPlayer_GetLastPlayedTimes_Response) body.getBody().build();
+            var gameIds = playedTimesResponse.getGamesList().stream()
+                    .map(game -> String.valueOf(game.getAppid()))
+                    .collect(Collectors.toList());
+            var predicate = getServiceMethodBodyPredicate(gameIds);
+
+            LOGGER.debug("Processing PlayedGame with matches: {}", gameIds);
+            findAndCompleteServiceMethodCallback(packetMessage.getMessageType().code(), CLIENT_APPLICATION_ID, packetMessage, predicate);
+        }
+    }
+
     private void handleGamePlayingSession(PacketMessage packetMessage) {
         ClientMessageProtobuf<CMsgClientPlayingSessionState.Builder> playingSessionBuilder = new ClientMessageProtobuf<>(
                 CMsgClientPlayingSessionState.class, packetMessage);
         var playingSession = playingSessionBuilder.getBody().build();
-        LOGGER.debug("Playing session game {} blocked: {}", playingSession.getPlayingApp(), playingSession.getPlayingBlocked());
-
-        Predicate<CompletableCallback> predicate = completableCallback -> Objects.nonNull(completableCallback.getProperties())
-                && String.valueOf(playingSession.getPlayingApp()).equals(
-                completableCallback.getProperties().get(PLAYER_LAST_PLAYED_TIMES).toString());
+        var predicate = getGamePlayedPredicate(playingSession);
```

Comment:
why u need to extract variable?

Reply:
For me `findAndCompleteServiceMethodCallback` method already has many parameters and using method as parameter will decrease readability. This part could be refactored if required.